### ISSUE TITLE
Instantaneous Push View Controller

### ISSUE THREAD

**Issue by willrax** [RubyCommotion/uikit-dynamics-catalog#3]

**Comment by dennismajor1** [comment]
I'll get tarted on Instantaneous Push View Controller

**Comment by willrax** [comment]
:+1:

**Comment by dennismajor1** [comment]
OK completed Instantaneous Push View Controller - I should have, I think it is 'squashed'?? some of the commit messages - however one of the recent four details what I did re the completion of this VC.

Interestingly enough, I was not blocked on the push which I think should have been the case since Will did some blank line cleanups between when I did my clone and my push yet there was no objection given when I did my push.

I did some cleanup myself so not sure if that would avoid the issue but I think it's determined by commit sequence - maybe you did a direct edit on the server file Will? Anyhow it's confusing to me that I wasn't blocked on my push.

**Comment by willrax** [comment]
Seems like you merged ok. Whats your usual process? Shouldn't need to have the merge commits at the top like "merged branch". Do you rebase your branch on top of master or do you just merge master in then merge your branch in to master?

**Comment by dennismajor1** [comment]
Yes the "merged branch" commit showed up because I forked to my github account and then added it as a remote to my local repo and was doing quick pushes to it as a means of backing up work. 

I think I should have done a rebase on my local repo as a final step before I then did a commit of the branch and merged it to the master.

So to answer your question I did a fork to my github account for backup purposes, I did a local git clone to start fresh locally, created a branch locally and worked on it. Additional  unnecessary commit messages came from pushing to my github account fork as a means of backing up work.

Bottom line I think I should have done a rebase at some point to clean up the commit messages :-) before pushing to origin??
